numpyro/infer/mcmc.py

```diff
@@ -752,10 +752,7 @@ def _single_chain_mcmc(self, rng_key, init_params, args, kwargs,
                                     diagnostics_fn=diagnostics)
         states, warmup_state = collect_vals
         # Get first argument of type `HMCState`
-        warmup_state = warmup_state[0]
-        # Note that setting i = 0 may result in recompilation due to python integers having
-        # weak type
-        self._warmup_state = warmup_state._replace(i=np.zeros_like(warmup_state.i))
+        self._warmup_state = warmup_state[0]
```

Comment:
@fehiepsi - I think this should be enough to get the correct behavior, but let me know if I missed something.